src/core/qgsexpressionlexer.ll

```diff
@@ -118,7 +118,7 @@ Q_GLOBAL_STATIC_WITH_ARGS(QLocale, cLocale, ("C") )
 
 %}
 
-%s BLOCK_COMMENT
+%x BLOCK_COMMENT
```

Comment:
For review purposes what's the difference between %s and %x?

Reply:
X means exclusive.

From the manual:

If the start condition is inclusive, then rules with no start conditions at all will also be active. If it is exclusive, then only rules qualified with the start condition will be active.